gossip/src/contact_info.rs

```diff
@@ -350,34 +350,42 @@ impl ContactInfo {
 
     pub fn set_socket(&mut self, key: u8, socket: SocketAddr) -> Result<(), Error> {
         sanitize_socket(&socket)?;
+        self.set_sockets_internal(key, &[socket])
+    }
+
+    fn set_sockets_internal(&mut self, key: u8, sockets: &[SocketAddr]) -> Result<(), Error> {
         // Remove the old entry associated with this key (if any).
         self.remove_socket(key);
-        // Find the index at which the new socket entry would be inserted into
-        // self.sockets, and the respective port offset.
-        let mut offset = socket.port();
-        let index = self.sockets.iter().position(|entry| {
-            offset = match offset.checked_sub(entry.offset) {
-                None => return true,
-                Some(offset) => offset,
+        for socket in sockets {
+            sanitize_socket(socket)?;
```

Comment:
would just be careful here that someone doesn't call set_sockets_internal somewhere in this file causing the socket to get removed but then fail sanitize. would suggesting something like:
```suggestion
    pub fn set_socket(&mut self, key: u8, socket: SocketAddr) -> Result<(), Error> {
        self.set_sockets_internal(key, &[socket])
    }

    fn set_sockets_internal(&mut self, key: u8, sockets: &[SocketAddr]) -> Result<(), Error> {
        for socket in sockets {
            sanitize_socket(socket)?;
        }   
        // Remove the old entry associated with this key (if any).
        self.remove_socket(key);
        
        for socket in sockets {
```

Reply:
i may have screwed up the formatting when I was trying to make a suggestion. but looking to do something like this:
```
pub fn set_socket(&mut self, key: u8, socket: SocketAddr) -> Result<(), Error> {
    self.set_sockets_internal(key, &[socket])
}
fn set_sockets_internal(&mut self, key: u8, sockets: &[SocketAddr]) -> Result<(), Error> {
    for socket in sockets {
        sanitize_socket(socket)?;
    }   
    // Remove the old entry associated with this key (if any).
    self.remove_socket(key);
    
    for socket in sockets {
        ...
...
```